pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -862,10 +946,44 @@ func (r *reconciler) Forget(ruleID string) error {
 	if r.fqdnController != nil {
 		r.fqdnController.deleteFQDNRule(ruleID, lastRealized.To.FQDNs)
 	}
+	if r.mcastController != nil {
+		groupAddresses, isIGMP := r.getMcastGroupAddress(lastRealized.CompletedRule)
+		klog.Infof("deleteing rule %v: groupAddresses %v, isIGMP %v", ruleID, groupAddresses, isIGMP)
+		if isIGMP {
+			r.mcastController.deleteGroupAddressForTableIDs(ruleID, groupAddresses)
+		}
+	}
 	r.lastRealizeds.Delete(ruleID)
 	return nil
 }
 
+func (r *reconciler) getMcastGroupAddress(rule * CompletedRule) ([]string, bool) {
+	groupAddresses := []string{}
+	isIGMP := false
+	if len(rule.Services) > 0 && (rule.Services[0].Protocol != nil) &&
+		(*rule.Services[0].Protocol == v1beta2.ProtocolIGMP) {
+		isIGMP = true
+		for _, service := range rule.Services {
+				if service.GroupAddress != nil {
+					groupAddressCidr := net.IPNet{
```

Comment:
use ip.IPNetToNetIPNet to transform IPBlock to net.IPNet

Reply:
changed groupAddress to IP, the transform is not need